tests/receivers/smartagent/postgresql/postgresql_test.go

```diff
@@ -15,117 +15,37 @@
 package tests
 
 import (
-	"context"
-	"fmt"
 	"path"
 	"testing"
-	"time"
-
-	"github.com/stretchr/testify/require"
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
-	"go.uber.org/zap/zaptest/observer"
 
 	"github.com/signalfx/splunk-otel-collector/tests/testutils"
 )
 
 func TestPostgresReceiverProvidesAllMetrics(t *testing.T) {
-	tc := newTestcase(t)
-	defer tc.printLogsOnFailure()
-	defer func() { require.NoError(t, tc.otlp.Shutdown()) }()
-
-	expectedResourceMetrics := tc.resourceMetrics("all.yaml")
-
-	server, client := tc.postgresContainers()
-	defer func() {
-		require.NoError(t, server.Stop(context.Background()))
-		require.NoError(t, client.Stop(context.Background()))
-	}()
-
-	collector := tc.splunkOtelCollector("all_metrics_config.yaml")
-	defer func() { require.NoError(tc, collector.Shutdown()) }()
-
-	require.NoError(t, tc.otlp.AssertAllMetricsReceived(t, *expectedResourceMetrics, 30*time.Second))
-}
-
-// While overkill for these test purposes, as a repeated pattern these should eventually be moved to testutils.
-type testcase struct {
-	*testing.T
-	logger       *zap.Logger
-	observedLogs *observer.ObservedLogs
-	otlp         *testutils.OTLPMetricsReceiverSink
-}
-
-func newTestcase(t *testing.T) *testcase {
-	tc := testcase{T: t}
-	var logCore zapcore.Core
-	logCore, tc.observedLogs = observer.New(zap.DebugLevel)
-	tc.logger = zap.New(logCore)
-
-	var err error
-	tc.otlp, err = testutils.NewOTLPMetricsReceiverSink().WithEndpoint("localhost:23456").WithLogger(tc.logger).Build()
-	require.NoError(tc, err)
-	require.NoError(tc, tc.otlp.Start())
-	return &tc
-}
+	containers := []testutils.Container{
+		testutils.NewContainer().WithContext(
+			path.Join(".", "testdata", "server"),
+		).WithEnv(map[string]string{
+			"POSTGRES_DB":       "test_db",
+			"POSTGRES_USER":     "postgres",
+			"POSTGRES_PASSWORD": "postgres",
+		}).WithExposedPorts(
+			"5432:5432",
+		).WithName("postgres-server").WithNetworks(
+			"postgres",
+		).WillWaitForPorts("5432").WillWaitForLogs(
+			"database system is ready to accept connections",
+		),
+		testutils.NewContainer().WithContext(
+			path.Join(".", "testdata", "client"),
+		).WithEnv(map[string]string{
+			"POSTGRES_SERVER": "postgres-server",
+		}).WithName("postgres-client").WithNetworks(
+			"postgres",
+		).WillWaitForLogs("Beginning psql requests"),
```

Comment:
This is difficult to read. I can't tell where slice items start and where they end. Or is it a one item? Hard to tell, looks like a wall of text.
Maybe try formatting differently or extract bits into variables.

Reply:
This better?  Unfortunately (imo) `gofmt` collapses things to one endless line without giving arguments their own individual line.